torch/nn/functional.py

```diff
@@ -5030,7 +5030,7 @@ def _scaled_dot_product_attention(
             have shape :math:`(B, Nt, Ns)`
     """
     B, Nt, E = q.shape
-    q = q / math.sqrt(E)
+    q /= math.sqrt(E)
```

Comment:
I would put a comment in the doc or next to the call site mentioning that. After calling this function, the q that was given should NOT be re-used anymore.